### ISSUE TITLE
[Usability] User experiences an unexpected interaction with a component.

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#77120]
_This ticket was generated during the pilot process for the new VA.gov experience standards. Please tag `@platform-governance-team-members` on Slack if you need assistance interpreting this ticket._

## Product Information
**Team:** Accredited Representation Management Team
**Product:** Find a Representative (MVP)
**Feature:** Find a Representative

## Findings details
**VA.gov Experience Standard - issue:** User experiences an unexpected interaction with a component.
**VA.gov Experience Standard - category:** Usability
**Launch-blocking:** Yes
**Collab Cycle Reviewer:** @briandeconinck (Accessibility)

### Description

When exiting the "Report outdated information" modal, keyboard focus doesn't go to an expected place. For keyboard-only users, the impact isn't too bad, because the next tab stop after closing the modal is still in the same card as where you started. But for screen reader users, keyboard focus also determines what's announced to them --- and right now, what's announced (the H2 heading) doesn't match where focus actually is.

## Recommended action
Here's what I recommend for focus management with these modals:
- When pressing "Submit," modal closes and focus goes to the `div` containing the "Thanks for reporting outdated information" alert (with focus set using `tabindex="-1"`. This will mean the first thing screen reader users hear when submitting outdated information is the alert text, so they know they've successfully completed the action.
- When pressing "Cancel" or the "X" icon to close the modal, focus returns to the "Report outdated information" button. That puts users back where they started, as if nothing has happened. (Because, in fact, nothing has happened!)

## References

- **WCAG Success Criteria:** [WCAG 1.3.2 Meaningful Sequence - Level A](https://www.w3.org/TR/WCAG21/#meaningful-sequence)
- **Modality:** N/A
- **Design System Component:** [Modal ](https://design.va.gov/components/modal)
- **Design System Pattern:** N/A
- **Content Style Guide:** N/A
- **Context:** N/A
- **Accessibility Defect Severity:** [1: Critical. Must be fixed before launch.](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric#Accessibilitydefectseverityrubric-a11y-defect-1)

---

### Next Steps for the VFS Team

- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members` with any questions or to get help validating the issue.
- [ ] **Ticket ownership:** Platform OCTO leadership asks that we document all issues we identify. If you believe that this issue is out of scope or not your team's responsibility, it is still your responsibility to resolve the issue. Please comment on Slack in your team channel tagging `@platform-governance-team-members`, providing an explanation and who you believe is responsible. Governance team will research and follow up.
- [ ] **Close this ticket** when the issue has been resolved or validated by your Product Owner.

**Comment by oddball-lindsay** [comment]
focus is looking good, but regression where unit test drops 0.8% below the acceptable threshold; need to add a check to avoid focus shifting to the wrong place.

we will get a PR in by tomorrow's review/release to address this, and can aim for beginning our staged rollout tomorrow towards the end of the day (after daily review/release).

**Comment by oddball-lindsay** [comment]
@briandeconinck we just completed our 508 accessibility audit of [Find a Representative](https://www.va.gov/get-help-from-accredited-representative/find-rep/) and it seems like setting the focus to tabindex="-1" (as recommended) is causing the "Report Outdated Information" to be skipped all together.  Now users relying on keyboard focus will miss out on the ability to interact with this feature because focus skips over that secondary button.

Do you have any alternative recommendations for improving the focus experience for screen reader users who have just submitted outdated information?

**Comment by oddball-lindsay** [comment]
Thanks for the heads up @shiragoodman!  I spoke with our a11y partner from CAIA (@coforma-jamie) and I think we just need to tweak the implementation originally put in place with PR https://github.com/department-of-veterans-affairs/vets-website/pull/28266.

Currently, the tabindex is paired with the `va-button` component, but Brian's recommendation is to place it with the `div` containing the "Thanks for reporting outdated information" alert.

![image](https://github.com/department-of-veterans-affairs/va.gov-team/assets/142453186/0e27a3af-bdad-42e7-b199-5d4c66ded0ed)

Hoping to tweak the implementation as Jamie recommends, and then we can verify that this resolves both
1. the original issue (focus goes to the "thank you for reporting outdated information" banner after user submits a report) 
2. the newly introduced issue (cannot place initial focus on the button)

**Comment by oddball-lindsay** [comment]
I see the PR merged and am able to tab to the "Report Outdated Information" button in search results now 👍
And when I cancel out of the modal, focus is back on the button 👍 👍

I think we're good here - closing out.  Please re-open if I misunderstood!
